src/DeckBuilder.tsx

```diff
@@ -55,52 +55,59 @@ async function download(form: any): Promise<string> {
         decklist = await getDeckFromURL(decklist[0]);
     }
 
+    // Process but do not commit commanders
+    [commander, partner].forEach((line: string, index: number) => {
+        if (isLineEmpty(line)) return;
+        let tmpCard = Card.fromLine(cleanLine(line));
+        tmpCard.setBackUrl(cardBack);
+        tmpCard.setCardType(CardType.Commander);
+        commanders.push(tmpCard);
+    });
+
     // Build decklist with queries
     decklist.forEach((line: string, index: number) => {
-        if (line === "" || line.startsWith("//")) {
-            return;
-        }
-        let tmpCard = Card.fromLine(line.trim());
+        if (isLineEmpty(line)) return;
+        let tmpCard = Card.fromLine(cleanLine(line));
         tmpCard.setBackUrl(cardBack);
-
-        if (commanderToBeHandled.length > 0) {
-            let isCommander = commanderToBeHandled.includes(getName(tmpCard.name));
-            if (isCommander) {
-                commanderToBeHandled.splice(commanderToBeHandled.indexOf(getName(tmpCard.name)), 1);
-                tmpCard.setCardType(CardType.Commander);
-                commanderIndices.push(index);
+        let isCommander = false;
+
+        // replace processed commanders when relevant
+        commanders = commanders.filter((commander: Card) => {
+            if (commander.name !== tmpCard.name) {
+                return true;
+            } else {
+                isCommander = true;
+                return false;
             }
+        });
+
+        // process special commander flag
+        if (line.includes("!Commander")) {
```

Comment:
maybe define a variable somewhere where you can set these flags?

Reply:
Like this?